.github/workflows/test.yaml

```diff
@@ -1,36 +1,38 @@
 # /.github/workflows/test.yaml
-# Build, lint and test the library
+# Test everything with the library
 
 name: Test
 
 on: [push, pull_request]
 
 jobs:
   test:
-    name: Build, Lint and Test
+    name: Test
     strategy:
       matrix:
-        node-version: [12.x, 14.x, 15.x, 16.x, 17.x]
-        os: [ubuntu-latest, macos-latest, windows-latest]
-    runs-on: ${{ matrix.os }}
-
+        node-version: [14.x, 15.x, 16.x, 17.x]
+    runs-on: ubuntu-latest
```

Comment:
Do you think we could run at least `npm run test:lib` on mac and windows? Maybe just on node 16.*. I think there's a way to add extra jobs to the matrix, but I don't remember what it is off the top of my head.

It'd be really nice if we could just define this once and then make the release.yml depend on the test.yml, but I don't think github supports that...

Reply:
Maybe we can have only one workflow file, `ci.yaml`, with the following jobs:

- `build` - runs `npm run compile`, makes sure that the code is valid typescript on node 14 to 17, and on linux, windows and macos
- `lint` - run `npm run lint`, makes sure there are no linting or formatting errors on node 14 to 17, and on linux, windows and macos
- `test_library`: run `npm run test:lib`, runs the tests for the library on node 14 to 17, and on linux, windows and macos
- `test_external`: run `npm run test: ext`, runs the tests that checks if the imports and external stores work on node 14 to 17, and on linux only (because the external store tests need docker containers running, this is supported by github actions only on linux runners)
- `release` - runs only if a tag is pushed (`if: startsWith(github.ref, 'refs/tags/v')`), needs the all the above jobs to succeed, and publishes the `npm pack`ed package to the registry. Also creates a new release.

Reply:
Added this in 32202bae817ef4e23d3294c96f4d4e6c61f22d6d 🚀